packages/core-data/src/resolvers.js

```diff
@@ -453,6 +453,19 @@ __experimentalGetTemplateForLink.shouldInvalidate = ( action ) => {
 	);
 };
 
+export const __experimentalGetCurrentSiteGlobalStylesId =
+	() =>
+	async ( { dispatch } ) => {
+		const siteGlobalStyles = await apiFetch( {
+			path: '/wp/v2/global-styles/',
```

Comment:
So, we need a way to pass the ID of the Custom Post Type that holds the data for the `site` origin (user changes unbound to a theme) to the client.

I see that the ID of the CPT for the `custom` origin (user changes bound to a theme) is taken from the [themes endpoint](https://github.com/WordPress/gutenberg/blob/trunk/packages/core-data/src/resolvers.js#L456) through the `wp:user-global-styles` property. It was introduced at https://github.com/WordPress/gutenberg/pull/35801. I see [a mention](https://github.com/WordPress/gutenberg/pull/35801#issuecomment-952190033) about how it could be extended in the future.

I wonder what our options are for this:

1. Attach it to the themes' endpoint as `wp:site-global-styles`, mimicking what we do with `wp:user-global-styles`. The data for the `site` is unrelated to the theme, so it feels wrong to do it this way.
2. Attach it as a link elsewhere: where?
3. Have a new endpoint to expose that piece of data. The current implementation of this PR. This seems too much considering the `custom` origin uses a link in an existing endpoint.

It seems 2 is more measured for what we're trying to do here, though we'd need to understand _where_. @youknowriad I'd like to hear your thoughts on this.

Reply:
> Attach it as a link elsewhere: where?

Perhaps a link in the [settings endpoint](https://developer.wordpress.org/rest-api/reference/settings/) that we already expose as the [site entity](https://github.com/WordPress/gutenberg/blob/trunk/packages/core-data/src/entities.js#L41)?